### ISSUE TITLE
{"error":"partial write: unable to parse ...

### ISSUE THREAD

**Issue by Dragonsys** [kernelkaribou/synology-monitoring#18]
Related to Issue #13 

I am still getting this issue when I include disk in the capture. I checked the above lines in the script and they already match what is listed as the fix. I have 14 disks (2 are for cache) in my enclosure

I get this error when trying to collect Raid or Disk info.

```
HTTP/1.1 400 Bad Request
Content-Type: application/json
Request-Id: 4774141a-3a4b-11eb-9423-0242ac110008
X-Influxdb-Build: OSS
X-Influxdb-Error: partial write: unable to parse 'synology_raid,nas_name=NAS,raid_name=Volume1 raid_status=,raid_free_size=,raid_total_size=': missing field value unable to parse 'synology_raid,nas_name=NAS,raid_name=StoragePool1 raid_status=,raid_free_size=,raid_total_size=': missing field value unable to parse 'synology_disk,nas_name=NAS,disk_name=Drive2 disk_model=SYNOLOGY-DISK-MIB::diskModel.13=STRING:"WDCWDS500G2B0C-00PXH0",disk_type=SYNOLOGY-DISK-MIB::diskType.13 = STRING: "SSD",disk_temp=SYNOLOGY-DISK-MIB::diskTemperature.13 = INTEGER: 41,disk_status=SYNOLOGY-DISK-MIB::diskStatus.13 = INTEGER: 1': invalid boolean dropped=0
X-Influxdb-Version: 1.8.3
X-Request-Id: 4774141a-3a4b-11eb-9423-0242ac110008
Date: Wed, 09 Dec 2020 18:20:56 GMT
Content-Length: 665

{"error":"partial write: unable to parse 'synology_raid,nas_name=NAS,raid_name=Volume1 raid_status=,raid_free_size=,raid_total_size=': missing field value\nunable to parse 'synology_raid,nas_name=NAS,raid_name=StoragePool1 raid_status=,raid_free_size=,raid_total_size=': missing field value\nunable to parse 'synology_disk,nas_name=NAS,disk_name=Drive2 disk_model=SYNOLOGY-DISK-MIB::diskModel.13=STRING:\"WDCWDS500G2B0C-00PXH0\",disk_type=SYNOLOGY-DISK-MIB::diskType.13 = STRING: \"SSD\",disk_temp=SYNOLOGY-DISK-MIB::diskTemperature.13 = INTEGER: 41,disk_status=SYNOLOGY-DISK-MIB::diskStatus.13 = INTEGER: 1': invalid boolean dropped=0"}
```

**Comment by Dragonsys** [comment]
I was able to fix disk info by making the same change as in the fix for #13, but on lines 246, 250, 254 & 258.
Still working on the Raid info

For some reason the raid info is coming out blank

```
synology_raid,nas_name=NAS,raid_name=Volume1 raid_status=,raid_free_size=,raid_total_size=
synology_raid,nas_name=NAS,raid_name=StoragePool1 raid_status=,raid_free_size=,raid_total_size=
```

I tried adding the same fix as the disk info, but it still comes back blank. If I run the snmpwalk command in the terminal it returns all the correct info.

```
SYNOLOGY-RAID-MIB::raidIndex.0 = INTEGER: 0
SYNOLOGY-RAID-MIB::raidIndex.1 = INTEGER: 1
SYNOLOGY-RAID-MIB::raidName.0 = STRING: "Volume 1"
SYNOLOGY-RAID-MIB::raidName.1 = STRING: "Storage Pool 1"
SYNOLOGY-RAID-MIB::raidStatus.0 = INTEGER: 1
SYNOLOGY-RAID-MIB::raidStatus.1 = INTEGER: 1
SYNOLOGY-RAID-MIB::raidFreeSize.0 = Counter64: 33893299716096
SYNOLOGY-RAID-MIB::raidFreeSize.1 = Counter64: 297795584
SYNOLOGY-RAID-MIB::raidTotalSize.0 = Counter64: 76767214657536
SYNOLOGY-RAID-MIB::raidTotalSize.1 = Counter64: 79966158979072
```

**Comment by Dragonsys** [comment]
Found it, I made a typo when I was adjusting the raid settings. So adding the fix from #13 worked for raid as well.
All my changes:
line 23: Add
 `snmp_com_string="" #Community String for your SMNP settings`

replace all instances of "public" with "$snmp_com_string"

line 203: Replace With
`if [[ $line == "SYNOLOGY-RAID-MIB::raidStatus.$id "* ]]; then`

Line 207: Replace With
`if [[ $line == "SYNOLOGY-RAID-MIB::raidFreeSize.$id "* ]]; then`

Line 211: Replace With
`if [[ $line == "SYNOLOGY-RAID-MIB::raidTotalSize.$id "* ]]; then`

Line 247: Replace With
`if [[ $line == "SYNOLOGY-DISK-MIB::diskModel.$id "* ]]; then`

Line 251: Replace With
`if [[ $line == "SYNOLOGY-DISK-MIB::diskType.$id "* ]]; then`

Line 255: Replace With
`if [[ $line == "SYNOLOGY-DISK-MIB::diskStatus.$id "* ]]; then`

Line 259: Replace With
`if [[ $line == "SYNOLOGY-DISK-MIB::diskTemperature.$id "* ]]; then`

[synology_snmp.txt](https://github.com/kernelkaribou/synology-monitoring/files/5683919/synology_snmp.txt)
